sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -1057,4 +1064,28 @@ class MapboxMap internal constructor(
    * @return True if terrain is enabled for given style and false otherwise.
    */
   override fun terrainEnabled() = terrainEnabled
+
+  internal fun setCameraAnimationPlugin(cameraAnimationsPlugin: CameraAnimationsPlugin?) {
+    cameraAnimationsPlugin?.let {
+      this.cameraAnimationsPlugin = WeakReference(it)
+    }
+  }
+
+  /**
+   * Get an instance of [CameraAnimationsPlugin].
+   * In most cases must not be called directly.
+   */
+  override fun getCameraAnimationPlugin() = cameraAnimationsPlugin?.get()
+
+  internal fun setGesturesAnimationPlugin(gesturesPlugin: GesturesPlugin?) {
```

Comment:
In `MapPluginRegistry` we are using `CopyOnWriteArrayList` to cache gesture and animation plugin instances, that said there might be multi gesture and animation plugins. In that case, it will overwrite the former ones.
How about add a API `addMapPluginRegistry` for `MapboxMap`?

Reply:
Good catch - now we apply extension functions to one concrete plugin.